src/python/pants/engine/legacy/graph.py

```diff
@@ -530,7 +531,9 @@ def hydrate_sources(
   """
   # TODO(#5864): merge the target's selection of --glob-expansion-failure (which doesn't exist yet)
   # with the global default!
-  path_globs = sources_field.path_globs.copy(glob_match_error_behavior=glob_match_error_behavior)
+  path_globs = dataclasses.replace(
```

Comment:
Just for me, can you explain the changes to this file and how they relate to the wider PR?

Reply:
Sure! So we need dataclasses to be ~immutable to work with V2. When you have an immutable object / record, the way you update it is by returning a new version of it with the modified fields. `dataclasses.replace()` will copy over every single attribute, minus whatever you change via kwargs.

We originally achieved this via our custom `.copy()` method through the `DatatypeMixin`. Now, we use the std lib's version.

Reply:
Thanks :smile: